File: R/inference.R
#' Run MCMC inference for a GeoMix model
#'
#' @param model An initialised (but not yet fitted) GeoMix model object.
#' @param n_samples Integer number of posterior samples to draw after warm-up.
#' @param n_warmup Integer number of warm-up / burn-in samples (discarded).
#' @param seed Integer random seed for reproducibility.
#' @return A posterior samples object (TODO: define format, e.g., \code{posterior::draws}).
#' @export
run_mcmc <- function(model, n_samples = 1000L, n_warmup = 500L, seed = 0L) {
  # TODO: implement MCMC inference (e.g., via Stan / rstan / cmdstanr)
  stop("run_mcmc() is not yet implemented.")
}

File: src/geomix/inference.py
"""Inference routines for the GeoMix model."""


def run_mcmc(model, n_samples: int = 1000, n_warmup: int = 500, seed: int = 0):
    """Run MCMC inference for a fitted GeoMix model.

    Parameters
    ----------
    model : GeoMix
        An initialised (but not yet fitted) GeoMix model instance.
    n_samples : int
        Number of posterior samples to draw after warm-up.
    n_warmup : int
        Number of warm-up / burn-in samples (discarded).
    seed : int
        Random seed for reproducibility.

    Returns
    -------
    dict
        ArviZ ``InferenceData`` object (or equivalent) containing posterior
        samples.
    """
    raise NotImplementedError("MCMC inference is not yet implemented.")
